cmd/flightctl-userinfo-proxy/main.go

```diff
@@ -52,8 +56,25 @@ func main() {
 		IdleTimeout:       60 * time.Second, // Time to keep idle connections open
 	}
 
-	if err := server.ListenAndServe(); err != nil {
-		log.Fatal("Server failed to start:", err)
+	// Handle graceful shutdown
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGHUP, syscall.SIGTERM, syscall.SIGQUIT)
+	defer cancel()
+
+	go func() {
+		<-ctx.Done()
+		log.Println("Shutdown signal received, initiating graceful shutdown")
+
+		shutdownCtx, shutdownCancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
this should take the parent ctx and this timeout works correctly we will exit sooner the server shuts down in time or we force after 30s
```suggestion
		shutdownCtx, shutdownCancel := context.WithTimeout(ctx, 30*time.Second)
```